packages/shell-api-types/bin/compile-shell-api-types.js

```diff
@@ -190,9 +190,20 @@ const loadAll = () => {
   const main = fs.readFileSync(path.join(yamlDir, 'main.yaml'));
   const mainLib = yaml.load(main);
   const FILES = fs.readdirSync(yamlDir).filter((s) => (/[A-Z]/.test( s[0])));
-  let exports = '\nexport default ShellApi;\n';
-  const types = ['const unknown = {\n  type: \'unknown\',\n  attributes: {}\n}'];
-  const typeConsts = ['unknown'];
+  let exports = '';
+  const types = [
+    'const unknown = {\n  type: \'unknown\',\n  attributes: {}\n}',
+    `const ShellApi = {
+    type: 'ShellApi',
+    hasAsyncChild: false,
+    attributes: {
```

Comment:
This irritates me slightly because we are defining these types in a separate place from where they are implemented (i.e. they exist on the ShellAPI object but are not getting generated from code. So if we update the top-level commands we have to do it in two places)

Reply:
Can't read the code anymore, you mean command parsing vs invoking?

Reply:
Ah sorry, went away after rebasing, I will make a new comment.